Bottom: cd0e23b59fb40b2cc4cce8969626e8f68593cd5a
Top:    b2aac8b30bd5886cb4f6d1d7c66893b8050f32a9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-12 01:22:10 +0100

Debug / attempt to fix DRC crash


---

diff --git a/src/find.c b/src/find.c
index 15e40a1..2bddffd 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2389,6 +2389,10 @@ IsLineInPolygon (LineType *Line, PolygonType *Polygon)
        Coord wid = (Line->Thickness + Bloat + 1) / 2;
        Coord x1, x2, y1, y2;
 
+       printf ("Line thickness is %i, Bloat is %i\n",
+               Line->Thickness, Bloat);
+       printf ("Testing with wid=%i\n", wid);
+
        x1 = MIN (Line->Point1.X, Line->Point2.X) - wid;
        y1 = MIN (Line->Point1.Y, Line->Point2.Y) - wid;
        x2 = MAX (Line->Point1.X, Line->Point2.X) + wid;
diff --git a/src/polygon.c b/src/polygon.c
index 67532a9..10f62ac 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1707,6 +1707,7 @@ bool
 IsRectangleInPolygon (Coord X1, Coord Y1, Coord X2, Coord Y2, PolygonType *p)
 {
   POLYAREA *s;
+  printf ("X1=%i, X2=%i, Y1=%i, Y2=%i\n", X1, X2, Y1, Y2);
   if (!
       (s = RectPoly (min (X1, X2), max (X1, X2), min (Y1, Y2), max (Y1, Y2))))
     return false;
